subsys/shell/shell.c

```diff
@@ -248,6 +248,13 @@ static bool tab_prepare(const struct shell *shell,
 	(void)shell_make_argv(argc, *argv, shell->ctx->temp_buff,
 			      CONFIG_SHELL_ARGC_MAX);
 
+	if (*argc > CONFIG_SHELL_ARGC_MAX) {
+		return false;
+	}
+
+	/* terminate arguments with NULL */
+	(*argv)[*argc] = NULL;
```

Comment:
do we need this? maybe early leave when argc exceeds is enough?

Reply:
Yes, this change is needed to avoid calling `strcmp` with `argv[argc]` as an argument when it has random data.